src/main/resources/avro/common.avdl

```diff
@@ -18,33 +337,112 @@ enum Strand {
+
+  If `referenceName` is null, this must not be null.
   */
-  string referenceName;
+  union { null, string } sequenceId = null;
 
   /**
-  The 0-based offset from the start of the forward strand for that reference.
-  Genomic positions are non-negative integers less than reference length.
+  The 0-based offset from the start of the forward strand for that sequence.
+  Genomic positions are non-negative integers less than sequence length.
   */
   long position;
 
   /**
-  Strand the position is associated with.
+  Strand the position is associated with. `POS_STRAND` represents the forward
+  strand, or equivalently the left side of a base, and `NEG_STRAND` represents
+  the reverse strand, or equivalently the right side of a base.
   */
   Strand strand;
 }
 
 /**
+A `Segment` is a range on a sequence, possibly including the joins at the
+sequence's ends. It does not include base data. (The bases for a sequence are
+available through the `getSequence()` API call.)
+
+In the sequence "GTGG", the segment starting at index 1 on the forward strand
+with length 2 is the "TG" on the forward strand. The length-2 segment starting
+at index 1 on the reverse strand is "AC", corresponding to the first two base
+pairs of the sequence, or the last two bases of the reverse complement.
+*/
+record Segment {
+  /**
+  The sequence ID and start index of this `Segment`. This base is always
+  included in the segment, regardless of orientation.
+  */
+  Position start;
+
+  /**
+  The length of this `Segment`'s sequence. If `start` is on the forward strand,
+  the `Segment` contains the range [`start.position`, `start.position` +
+  `length`). If `start` is on the reverse strand, the `Segment` contains the
+  range (`start.position` - `length`, `start.position`]. This is equivalent to
+  starting from the side indicated by `start`, and traversing through that base
+  out to the specified length.
+  */
+  long length;
+
+  /**
+  Start and end `Position`s where this `Segment` attaches to other sequences.
+  Note that the segmentId for start and end might not be the same. The
+  `Segment`s covering the sequences onto which this `Segment` is joined are
+  called its "parents", while this segment is a "child".
+
+  Joins may occur on the outer sides of the terminal bases in a sequence: the
+  left side of the base at index 0, and the right side of the base with maximum
+  index. These are the "terminal sides" of the sequence. `startJoin` is the join
+  on the side indicated by `start`, and may only be set if that side is a
+  terminal side. Similarly, `endJoin` is the join on the opposite side of the
+  piece of sequence selected by the segment, and may only be set if that side is
+  a terminal side. The value of `startJoin` or `endJoin`, if set, is the side to
+  which the corresponding side of this `Sequence` is connected.
+  */
+  union { null, Position } startJoin, endJoin;
+}
+
+/**
+A `Path` is an ordered list of `Segment`s. In general any contiguous path
```

Comment:
The `Segment` order in a `Path` is the order in which you see those `Segment`s when walking from one end of the `Path` to the other. I can clarify that in the comment.

Unfortunately, we don't let you just take `Segment`s of a single `Sequence` and throw them into a `Path` in whatever order you want. See the bit about "no novel adjacencies"?

There needs to be a path in the sequence graph that your `Path` object is following, so if you want to see B, then A, then C, you need to add new (0-length) sequences for the novel adjacencies in that translocation, and have (0-length) `Segment`s in your `Path` saying that you follow those sequences. So it would really be ` { B, B-to-A adjacency, A, A-to-C adjacency, C}`.